Set EnableFrameworkPathOverride to false when targeting NETStandard or NETCore

jeffkl opened the pull request:
Resolving FrameworkPathOverride slows down evaluation and is not needed for NETStandard and NETCore.

Related to https://github.com/Microsoft/msbuild/pull/2583

davkean commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets:
```diff
@@ -62,6 +62,10 @@ Copyright (c) .NET Foundation. All rights reserved.
+
+    <!-- Disable the use of FrameworkPathOverride in Microsoft.Common.CurrentVersion.targets which can slow down evaluation.  FrameworkPathOverride
+    is not needed for NETStandard or NETCore since references come from NuGet packages-->
+    <UseFrameworkPathOverride>false</UseFrameworkPathOverride>
```
We let the user/other targets opt out, can you condition it against itself?

jeffkl replied:
I've added a condition

nguerrera commented:
Do we need to get this in 15.5?

davkean commented:
Yes for 15.5. Is there a reason we wouldn't take it?

nguerrera commented:
No, this is targeting the wrong branch is all. Release/15.5 please.

davkean commented:
@jeffkl I tried to retarget but you are currently on top of master so you'll need to rebase this against release/15.5.

jeffkl committed:
Set EnableFrameworkPathOverride to false when targeting NETStandard or NETCore

Resolving FrameworkPathOverride slows down evaluation and is not needed for NETStandard and NETCore.

jeffkl commented:
I have merged the corresponding change into MSBuild

https://github.com/Microsoft/msbuild/pull/2583

nguerrera commented:
@dsplaisted

dsplaisted reviewed (approved):
Looks good.  Once we have both the changes merged together (ie in a build of the .NET Core SDK), I'd like us to double-check that everything is working correctly together and we're no longer evaluating the framework path unnecessarily.

dsplaisted commented:
@MattGertz for approval.  See also VSO bug: https://devdiv.visualstudio.com/DevDiv/_workitems?id=507607&_a=edit

**Customer scenario**

Developing .NET Core or .NET Standard projects (Improves performance)

**Bugs this fixes**

https://github.com/Microsoft/msbuild/issues/2518

**Workarounds, if any**

n/a

**Risk**

Low

**Performance impact**

This will improve performance

**Root cause analysis**

MSBuild was calculating the `FrameworkPathOverride` values for all projects, even for .NET Core and .NET Standard projects which later reset the value to empty

**How was the bug found?**

[Evaluation Profiling](https://github.com/Microsoft/msbuild/blob/exp/EvaluationProfiler/documentation/evaluation-profiling.md)

MattGertz commented:
Please list out the actual perf win, but otherwise consider this approved to take to Shiproom.
